backend/danswer/server/search_backend.py

```diff
@@ -49,36 +46,20 @@ def read_server_status():
 
 @router.post("/direct-qa", response_model=QAResponse)
 def direct_qa(question: QAQuestion):
-    prompt_processor = BASIC_QA_PROMPTS[DEFAULT_PROMPT]
+    qa_model = get_default_backend_qa_model()
     query = question.query
     collection = question.collection
 
     logger.info(f"Received semantic query: {query}")
     start_time = time.time()
 
     ranked_chunks = semantic_search(collection, query)
-    sem_search_time = time.time()
-    top_docs = [ranked_chunk.document_id for ranked_chunk in ranked_chunks]
-    top_contents = [ranked_chunk.content for ranked_chunk in ranked_chunks]
 
-    logger.info(f"Semantic search took {sem_search_time - start_time} seconds")
-    files_log_msg = f"Top links from semantic search: {', '.join(top_docs)}"
-    logger.info(files_log_msg)
+    answer, quotes = qa_model.answer_question(query, ranked_chunks)
 
-    qa_answer = answer_question(query, top_contents, prompt_processor)
-    qa_time = time.time()
-    logger.debug(qa_answer)
-    logger.info(f"GPT QA took {qa_time - sem_search_time} seconds")
+    logger.info(f"Total QA took {time.time() - start_time} seconds")
 
-    # Postprocessing, no more models involved, purely rule based
-    answer, quotes_dict = process_answer(qa_answer, ranked_chunks)
-    postprocess_time = time.time()
-    logger.info(f"Postprocessing took {postprocess_time - qa_time} seconds")
-
-    total_time = time.time() - start_time
-    logger.info(f"Total QA took {total_time} seconds")
-
-    qa_response = {"answer": answer, "quotes": quotes_dict}
+    qa_response = {"answer": answer, "quotes": quotes}
```

Comment:
Would probably be slightly better to be: `QAResponse(answer=answer, quotes=quotes)`

Reply:
true true, good catch